### PR TITLE
INTMDB-553: L2 construct for ThirdPartyIntegration

### PR DESCRIPTION
**Pull request by maastha**
## Description

Added L2 construct over existing L1 CDK ThirdPartyIntegrations resource. 
Created 3 separate L2 constructs for integration with each of MicrosoftTeams, Datadog and PagerDuty under one **@mongodbatlas-awscdk/atlas-integrations** package.

Example usage:
```
import * as l2_tpi from '@mongodbatlas-awscdk/atlas-integrations';

new l2_tpi.DatadogIntegration(this, 'TPI_L2', {
      apiKeys: {
        publicKey: env.PUBLIC_KEY,
        privateKey: env.PRIVATE_KEY
      },
      projectId: "....",
      apiKey: ".....",
      region: l2_tpi.DatadogRegion.US,
    });
```

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by andreaangiolillo — APPROVED** [review]
LGTM once vulnerabilities are fixed

File: cdk/l2-cdk-resources/third-party-integration/src/util.ts
Comment (maastha) [reply cdk/l2-cdk-resources/third-party-integration/src/util.ts]: updated

**Review by andreaangiolillo — APPROVED** [review]

File: cdk/l2-cdk-resources/third-party-integration/test/constants.ts
Comment (andreaangiolillo) [inline cdk/l2-cdk-resources/third-party-integration/test/constants.ts:21]: 👀

File: cdk/l2-cdk-resources/third-party-integration/.projenrc.js
Comment (andreaangiolillo) [inline cdk/l2-cdk-resources/third-party-integration/.projenrc.js:35]: 👀

File: cdk/l2-cdk-resources/third-party-integration/src/thirdPartyIntegrationBase.ts
Comment (SuperMohit) [inline cdk/l2-cdk-resources/third-party-integration/src/thirdPartyIntegrationBase.ts:37]: would we remove these later and use the profile?

**Comment by SuperMohit** [comment]
Do we need both yarn and package for version locking? If No, then we can remove package.lock.

**Review by SuperMohit — APPROVED** [review]

File: cdk/l2-cdk-resources/third-party-integration/.projenrc.js
Comment (colm-quinn) [reply cdk/l2-cdk-resources/third-party-integration/.projenrc.js:35]: I think all the tools aren't adding newlines, which is a little annoying.

Also, do we need to commit the generated code or can that be generated via the makefile as part of scripting when compiling/testing/publishing?

Better way of asking the question maybe, do we edit these files after the initial generation?

File: cdk/l2-cdk-resources/third-party-integration/src/util.ts
Comment (colm-quinn) [inline cdk/l2-cdk-resources/third-party-integration/src/util.ts:32]: [nit] Nice to put all validation for an object (in this case `props.apiKeys`) in its own helper method, so it can be reused & modified in isolation.

File: cdk/l2-cdk-resources/third-party-integration/yarn.lock
Comment (colm-quinn) [inline cdk/l2-cdk-resources/third-party-integration/yarn.lock:1]: Do we need this file, if `package.lock` already exists? We've had issues previously with libraries that scan the codebase when both package managers are present.

Single source of truth is best, if possible.

File: cdk/l2-cdk-resources/third-party-integration/.projenrc.js
Comment (andreaangiolillo) [reply cdk/l2-cdk-resources/third-party-integration/.projenrc.js]: yes, we have to edit `.projenrc.js` after it is generated

File: cdk/l2-cdk-resources/third-party-integration/test/constants.ts
Comment (maastha) [reply cdk/l2-cdk-resources/third-party-integration/test/constants.ts]: updated

File: cdk/l2-cdk-resources/third-party-integration/.projenrc.js
Comment (maastha) [reply cdk/l2-cdk-resources/third-party-integration/.projenrc.js]: For other generated files, I think since currently we publish manually, it's okay because we'll have tested code in the repo.

Comment (maastha) [reply cdk/l2-cdk-resources/third-party-integration/.projenrc.js]: updated new line

File: cdk/l2-cdk-resources/third-party-integration/src/util.ts
Comment (maastha) [reply cdk/l2-cdk-resources/third-party-integration/src/util.ts]: added separate helper method.

File: cdk/l2-cdk-resources/third-party-integration/yarn.lock
Comment (maastha) [reply cdk/l2-cdk-resources/third-party-integration/yarn.lock:1]: removed package-lock

**Comment by maastha** [comment]
> Do we need both yarn and package for version locking? If No, then we can remove package.lock.

Removed package-lock.

File: cdk/l2-cdk-resources/third-party-integration/src/thirdPartyIntegrationBase.ts
Comment (maastha) [reply cdk/l2-cdk-resources/third-party-integration/src/thirdPartyIntegrationBase.ts:37]: Yes, that can be addressed later with other constructs.

**Review by colm-quinn — APPROVED** [review]
LGTM, thanks for the updates.

File: cdk/l2-cdk-resources/third-party-integration/.projenrc.js
Comment (maastha) [reply cdk/l2-cdk-resources/third-party-integration/.projenrc.js]: For Typescript, I've installed Prettier for now in VS code, that should help in the future

Comment (colm-quinn) [reply cdk/l2-cdk-resources/third-party-integration/.projenrc.js]: Nice. Not blocking this PR, but also check out how we enforce TS code formatting in MMS - it's a similar use-case we've solved.

A good candidate for a small improvement in the future.

**Review by Zuhairahmed — APPROVED** [review]
LGTM @maastha!

**Pull request merged**
